frontend/catalyst/compilation_pipelines.py

```diff
@@ -837,6 +849,22 @@ def qjit(
             elements of this list are named sequences of MLIR passes to be executed. A ``None``
             value (the default) results in the execution of the default pipeline. This option is
             considered to be used by advanced users for low-level debugging purposes.
+        abstracted_axes (Optional(Any)): Experimental, a way to specify dynamic tensor shapes.
+            There are two ways to use abstracted_axes, one is passing a sequence of tuples and the
+            other is passing a dictionary. When passing a sequence of tuples
+
+                abstracted_axes=(('n',), ('m', 'n'))
+
+            The intended meaning is that the first argument will have the 0th axis be the symbolic
+            variable "n" and the second argument will have the 0th axis be the symbolic variable
+            "n" and the 1st axist will be the symbolic vairable "m".
+
+            When passing a dictionary:
+
+                abstracted_axes={0: 'n'}
+
+            it specifies that for all parameters, the 0th axis will be the symbolic variable "n".
+            This allows the user to express relationships between axes in different parameters.
```

Comment:
```suggestion
        abstracted_axes (Sequence[tuple] or dict): Experimental, a way to specify dynamic tensor shapes.
            There are two ways to use ``abstracted_axes``, one is passing a sequence of tuples and the
            other is passing a dictionary. When passing a sequence of tuples,

            .. code-block:: python

                abstracted_axes=(('n',), ('m', 'n'))

            the intended meaning is that:
            
            - the first argument will have the zeroth axis denoted by ``n``, and
            
            - the second argument will have the 0th axis denoted by
              ``m`` and the first axis denoted by ``n``.

            When passing a dictionary:

            .. code-block:: python

                abstracted_axes={0: 'n'}

            it specifies that for all parameters, the zeroth axis will be denoted by ``n``.
            This allows the user to express relationships between axes in different parameters.
```

Reply:
There are three things I don't understand here.

1. What happens if you want to set abstracted axis for some function arguments, but not all? E.g., you want to set it for the 3rd and 5th function argument only?

2. It's actually not clear to me what the `abstracted_axis` parameter corresponds to. Function arguments of the decorated function? Then what do you mean when you say 'parameters' in the second last sentence? Should this be changed to 'it specifies that for all _function arguments_, ...'

3. Finally, I'm not sure I understand the feature this kwarg unlocks :) Why would I want to use this? When would I _need_ to use this? An example (not here, but in the example section below) could help.

Reply:
Thanks @erick-xanadu, this is helpful! I suggest changing parameters -> function arguments, as this is a bit more consistent language.

One thing I am still confused about: when will this help me? That is, it would be super helpful to see an example that shows:

- calling a function with arrays with changing shape, causing re-compilation
- Next, using `abstracted_axis` to avoid this issue, and showing calling a compiled function with tensors of different shapes, and showing there is no recompilation.

Reply:
Ha! This was a good test and thanks for pointing it out. I just added the functionality to avoid recompilation in this case. See the documentation here: https://github.com/PennyLaneAI/catalyst/pull/366/commits/bd46085ed2a427538de8ffc5bcf3b4cde59892b4 Please resolve this comment if you think this has answered your questions.

Reply:
thanks Erick :)